franklin/builder/models.py

```diff
@@ -92,40 +89,62 @@ class Build(models.Model):
 
     :param site: Reference to the project for this build instance
     :param git_hash: If a branch build, the git hash of the deployed code
-    :param branch: If a branch build, the name of the branch
-    :param tag: If a tag build, the name of the tag
     :param created: Date this code was built
     :param path: The path of the site on the static server
     """
 
     site = models.ForeignKey(Site, related_name='builds')
-    git_hash = models.CharField(max_length=40, blank=True, null=True)
-    branch = models.CharField(max_length=100, blank=True, null=True)
-    tag = models.CharField(max_length=100, blank=True, null=True)
     created = models.DateTimeField(editable=False)
```

Comment:
Refactoring this to use inheritance since `TagBuild` and `BranchBuild` are different flavors of a build. Code is cleaner, only issue is that, behind the scenes, this creates some extra table joins which may result in more DB calls in the future. Not sure I'm worried about that though.

Let's discuss